src/main/python/ayab/engine/engine_fsm.py

```diff
@@ -56,13 +63,14 @@ class StateMachine(object):
     @date   June 2020
     """
 
-    def set_transitions(self, parent):
+    def set_transitions(self, parent:Engine)->None:
         """Define transitions between states for Finite State Machine"""
 
         # Events that trigger state changes
-        self.CONNECT.addTransition(parent.port_opener, self.VERSION_CHECK)
+        self.CONNECT.addTransition(parent.port_opener, self.VERSION_CHECK) #type: ignore
```

Comment:
I honestly don't understand how these work, so I couldn't really figure out how to fix this typing. There's a reason Magick™️  is falling out of favour 😛

Reply:
I fixed the typing by explicitly declaring `CONNECT` and `VERSION_CHECK` as `QState`s, I'm not sure I believe they _ARE_ (I have _NO_ idea where that hook-up happens) but it means this is not a type error anymore.